lib/gulpfile.js

```diff
@@ -0,0 +1,207 @@
+    'hide_title: false\n' +
+    'hidden: false\n' +
+    'bootstrap: false\n' + 
+    '#excerpt_separator: <!--more-->\n' +
+    '---'+
+    '\nWrite here...';
+  console.log('[' + new Date().toLocaleTimeString('en-CA', {hour12: false}) + '] File created: _posts/' + filename);
+  fs.writeFile(postsPath + filename, content, callback);
+}
+
+// Minify JS
+const js = function(cb) {
+  pump([
+     src(paths.js.src),
+     concat('main.min.js'),
+     uglify({output: {comments: 'some'}}), //will preserve multi-line comments w/ @preserve, @license or @cc_on
+     dest(paths.js.dest)
+  ],
+  cb()
+  );
+}
+
+// TODO: Updated Bootstrap to 4.6
+// Isolate Bootstrap
+const bsIsolate = function(cb) {
+  pump([
+     src(paths.css.src+'bootstrap-iso.less'),
+     less({strictMath: 'on'}),
+     replace('.bootstrap-iso html', ''),
+     replace('.bootstrap-iso body', ''),
+     dest(paths.css.src)
+  ],
+  cb()
+  );
+}
+
+// Minify Bootstrap CSS
+const bsMinify = function(cb) {
+  pump([
+     src(paths.css.src+'bootstrap-iso.css'),
+     cleanCSS(),
+     concat('bootstrap-iso.min.css'),
+     dest(paths.css.dest)
+  ],
+  cb()
+  );
+}
+
+// Resize and Minify IMG
+const imgAvatar = function(cb) {
+  pump([
+     src(paths.avatar.src),
+     changed(paths.avatar.dest),
+     responsive({'*': {width: 512}}),
+     imagemin({verbose: true}),
+     dest(paths.avatar.dest)
+  ],
+  cb()
+  );
+}
+
+const imgFeatured = function(cb) {
+  pump([
+     src(paths.featured.src),
+     changed(paths.featured.dest),
+     responsive({'*': {width: 1920}}),
+     imagemin({verbose: true}),
+     dest(paths.featured.dest)
+  ],
+  cb()
+  );
+}
+
+const imgThumbnails = function(cb) {
+  pump([
+     src(paths.thumbnails.src),
+     changed(paths.thumbnails.dest),
+     responsive({'*': {width: 1024}}),
+     dest(paths.thumbnails.dest)
+  ],
+  cb()
+  );
+}
+
+const imgPosts = function(cb) {
+  pump([
+     src(paths.posts.src),
+     changed(paths.posts.dest),
+     //responsive({'*': {width: 900}}),
+     imagemin({verbose: true}),
+     dest(paths.posts.dest)
+  ],
+  cb()
+  );
+}
+
+const imgPortfolio = function(cb) {
+  pump([
+     src(paths.portfolio.src),
+     changed(paths.portfolio.dest),
+     responsive({'*': {width: 900}}),
+     imagemin({verbose: true}),
+     dest(paths.portfolio.dest)
+  ],
+  cb()
+  );
+}
+
+const imgWebp = function(cb) {
+  pump([
+     src(paths.webp.src),
+     changed(paths.webp.dest),
+     webp({quality: 85, preset: 'photo', method: 6}),
+     dest(paths.webp.dest)
+  ],
+  cb()
+  );
+}
+
+// Tasks
+exports.post = post;
+exports.js = js;
+exports.bootstrap = series(bsIsolate, bsMinify);
+
+exports.avatar = imgAvatar;
+exports.featured = imgFeatured;
+exports.thumbs = imgThumbnails;
+exports.posts = imgPosts;
+exports.portfolio = imgPortfolio;
+exports.webp = imgWebp;
+exports.img = parallel(imgAvatar, imgPortfolio, series(imgFeatured, imgThumbnails));
+
+// FIXME: Gulp will generate thumbnails only on 2nd run
```

Comment:
Why is there a FIXME ?

Reply:
If I put an image at `_ img/featured` and them run gulp it will generate a converted and minified image at `img/featured`. If I run gulp by a second time it will generate a thumbnail of the converted image at `img/thumbnails/featured/`. Those tasks are serialized so both should be generated on first run. But  gulp or one of it's modules (changed or responsive) scopes the files at the beginning causing this behaviour. It's a non-critical bug.

Reply:
Oh I see due to the fact that there's two folders, you first need to generate the compress image before it can actually be made into a thumbnail. That's unfortunate.

Reply:
Problem is gulp being so fragile. Some ideas:
1. I could generate thumbnails in parallel from the source images. Problem is: what if there`s no source image?
2. I should modify the include to automatically use the thumbnail if it exists or else revert to the image at `../featured`. The user should not have to waste time with those details.

Reply:
Long term I am tempted to write my own helper. Something simple, just for Jekyll, to automate those nuissances. But I have to find time for that.